### PR TITLE
Swagger Documentation for Outbound Firewall Rules

### PR DESCRIPTION
**Pull request by VMMicrosoft**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Please ensure to add changelog with this PR by answering the following questions.
  1. What's the purpose of the update?    
      - [ ] new service onboarding 
      - [x] new API version 
      - [ ] update existing version for new feature 
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify 
  2. When you are targeting to deploy new service/feature to public regions? Please provide date, or month to public if date is not available yet.
  3. When you expect to publish swagger? Please provide date, or month to public if date is not available yet.
  4. If it's an update to existing version,  please select SDKs of specific language and CLIs that require refresh after swagger is published.
      - [x] SDK of .NET (need service team to ensure code readiness)
      - [x] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [x] PowerShell
      - [x] CLI
      - [ ] Terraform
      - [ ] No, no need to refresh for updates in this PR

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [x] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  -  [ ] Ensure to copy the existing version into new directory structure for first commit (including refactoring) and then push new changes including version updates in separate commits. This is required to review the changes efficiently.
  - Adding a new service

- [ ] Please ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Review by ericshape — CHANGES_REQUESTED** [review]

**Comment by ericshape** [comment]
feel free to ignore these LintDiff errors. SQL DB does not support it until now.

Rule | Message
-- | --
❌ R4009 - RequiredReadOnlySystemData | The response of operation:'OutboundFirewallRules_Get' is defined without 'systemData'. Consider adding the systemData to the response.Location: Microsoft.Sql/preview/2021-02-01-preview/OutboundFirewallRules.json#L20
❌ R4009 - RequiredReadOnlySystemData | The response of operation:'OutboundFirewallRules_CreateOrUpdate' is defined without 'systemData'. Consider adding the systemData to the response.Location: Microsoft.Sql/preview/2021-02-01-preview/OutboundFirewallRules.json#L63

**Comment by ericshape** [comment]
the .net sdk errors are expected due to we have v5 sdk tag. and .net sdk has breaking changes.

File: specification/sql/resource-manager/Microsoft.Sql/preview/2021-02-01-preview/examples/OutboundFirewallRuleGet.json [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    "api-version": "2021-02-01-preview"
+  },
+  "responses": {
+    "200": {
+      "body": {
+        "id": "/subscriptions/00000000-1111-2222-3333-444444444444/resourceGroups/sqlcrudtest-7398/providers/Microsoft.Sql/servers/sqlcrudtest-4645/outboundFirewallRules/server.datbase.windows.net",
+        "name": "server.database.windows.net",
+        "type": "Microsoft.Sql/servers/outboundFirewallRules",
+        "properties": {
+          "provisioningState": "Ready"
```
Comment (zcxcasd) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2021-02-01-preview/examples/OutboundFirewallRuleGet.json:16]: >"Ready" [](start = 31, length = 7)

Is this a valid provisioningState? Maybe Succeeded?

File: specification/sql/resource-manager/Microsoft.Sql/preview/2021-02-01-preview/OutboundFirewallRules.json [hunk 2]
```diff
@@ -0,0 +1,284 @@
...
+  "definitions": {
+    "OutboundFirewallRuleProperties": {
+      "description": "The properties of an outbound firewall rule.",
+      "type": "object",
+      "properties": {
+        "provisioningState": {
+          "description": "The state of the outbound rule.",
+          "type": "string",
+          "readOnly": true
+        }
```
Comment (zcxcasd) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2021-02-01-preview/OutboundFirewallRules.json:224]: note that properties.provisioningState cannot be any string, it needs to follow rpc.

File: specification/sql/resource-manager/Microsoft.Sql/preview/2021-02-01-preview/examples/ServerList.json [hunk 3]
```diff
@@ -19,6 +19,8 @@
               "version": "12.0",
               "state": "Ready",
               "workspaceFeature": "Connected",
+              "publicNetworkAccess": "Enabled",
```
[the remark this answers is not in the document]
Comment (VMMicrosoft) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2021-02-01-preview/examples/ServerList.json:22]: Done already.

File: specification/sql/resource-manager/readme.md [hunk 4]
```diff
@@ -108,6 +108,7 @@ input-file:
   - Microsoft.Sql/preview/2021-02-01-preview/ManagedServerSecurityAlertPolicies.json
   - Microsoft.Sql/preview/2021-02-01-preview/Operations.json
   - Microsoft.Sql/preview/2021-02-01-preview/OperationsHealth.json
+  - Microsoft.Sql/preview/2021-02-01-preview/OutboundFirewallRules.json
```
[the remark this answers is not in the document]
Comment (VMMicrosoft) [reply specification/sql/resource-manager/readme.md:111]: Added.

**Review by ericshape — APPROVED** [review]
LGTM

All the failed checks are expected.
1. the systemData lint diff errors are expected. 
2. the .net SDK errors are also expected, @viparek is working on it now. 

@akning-ms  can you review and merge this PR to the dev branch? Thanks!

File: specification/sql/resource-manager/Microsoft.Sql/preview/2021-02-01-preview/Servers.json [hunk 5]
```diff
@@ -523,7 +523,7 @@
           ],
           "type": "string",
           "x-ms-enum": {
-            "name": "ServerPublicNetworkAccess",
+            "name": "ServerNetworkAccessFlag",
```
Comment (akning-ms) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2021-02-01-preview/Servers.json:526]: What is the reason to rename this enum name? as it create SDK breaking change

Comment (beoberha) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2021-02-01-preview/Servers.json:526]: The enum is being used in more places with this change, so we renamed it to make it more generic.

**Pull request merged**